src/components/frame/v5/pages/AgreementsPage/partials/AgreementCard/AgreementCard.tsx

```diff
@@ -0,0 +1,248 @@
+import { MotionState as NetworkMotionState } from '@colony/colony-js';
+import clsx from 'clsx';
+import { format } from 'date-fns';
+import { BigNumber } from 'ethers';
+import { FilePlus, ShareNetwork } from 'phosphor-react';
+import React, { useState, type FC, useEffect, useMemo } from 'react';
+import { generatePath, useNavigate } from 'react-router-dom';
+
+import MeatballMenuCopyItem from '~common/ColonyActionsTable/partials/MeatballMenuCopyItem/MeatballMenuCopyItem.tsx';
+import { useColonyContext } from '~context/ColonyContext.tsx';
+import { useMobile } from '~hooks/index.ts';
+import useUserByNameOrAddress from '~hooks/useUserByNameOrAddress.ts';
+import {
+  COLONY_AGREEMENTS_ROUTE,
+  COLONY_HOME_ROUTE,
+  TX_SEARCH_PARAM,
+} from '~routes';
+import { getMotionState, MotionState } from '~utils/colonyMotions.ts';
+import { getEnumValueFromKey } from '~utils/getEnumValueFromKey.ts';
+import { formatText } from '~utils/intl.ts';
+import { getSafePollingInterval } from '~utils/queries.ts';
+import { useGetColonyAction } from '~v5/common/ActionSidebar/hooks/useGetColonyAction.ts';
+import MotionCountDownTimer from '~v5/common/ActionSidebar/partials/Motions/partials/MotionCountDownTimer/MotionCountDownTimer.tsx';
+import {
+  type Steps,
+  CustomStep,
+} from '~v5/common/ActionSidebar/partials/Motions/types.ts';
+import MotionStateBadge from '~v5/common/Pills/MotionStateBadge/MotionStateBadge.tsx';
+import TeamBadge from '~v5/common/Pills/TeamBadge/TeamBadge.tsx';
+import Avatar from '~v5/shared/Avatar/Avatar.tsx';
+import MeatBallMenu from '~v5/shared/MeatBallMenu/MeatBallMenu.tsx';
+import UserPopover from '~v5/shared/UserPopover/UserPopover.tsx';
+
+import AgreementCardSkeleton from '../AgreementCardSkeleton.tsx';
+
+import { type AgreementCardProps } from './types.ts';
+
+const AgreementCard: FC<AgreementCardProps> = ({ transactionId }) => {
+  const isMobile = useMobile();
+  const {
+    action,
+    refetchMotionState,
+    loadingAction,
+    motionState,
+    startPollingForAction,
+    stopPollingForAction,
+  } = useGetColonyAction(transactionId);
+  const { decisionData, motionData } = action || {};
+  const {
+    createdAt = '',
+    description = '',
+    title,
+    motionDomainId,
+    walletAddress = '',
+  } = decisionData || {};
+  const { motionId = '', motionStakes, requiredStake = '' } = motionData || {};
+  const navigate = useNavigate();
+  const { colony } = useColonyContext();
+  const { user, loading } = useUserByNameOrAddress(walletAddress);
+  const currentTeam = colony?.domains?.items.find(
+    (domain) => domain?.nativeId === motionDomainId,
+  );
+
+  // @TODO: Change this logic when https://github.com/JoinColony/colonyCDapp/pull/1842 is implemented
```

Comment:
It has been merged. `useGetColonyAction` now exports `motionState` and `networkMotionState`. You can reference `Motions.tsx` to see the new usage. You shouldn't need `getEnumValueFromKey` or `motionStateEnum` transformation at all.

Reply:
@jakubcolony I'm not using `--no-verify` option, but I think there are no more build errors